pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -68,14 +75,26 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 			// This is a change to a configmap we don't watch, ignore it
 			return false
 		}
+		if features.EnableEnhancedResourceScoping && !c.DiscoveryNamespacesFilter.Filter(o.GetNamespace()) {
```

Comment:
```suggestion
		if features.EnableEnhancedResourceScoping && !c.DiscoveryNamespacesFilter.Filter(o) {
```

Wouldn't this work as well without the need to add the new case in Filter to handle a string?

Reply:
added separate FilterNamespace()